src/mscorlib/corefx/System/Globalization/CalendarData.Unix.cs

```diff
@@ -193,8 +321,34 @@ private static bool EnumCalendarInfo(string localeName, CalendarId calendarId, C
 
         private static void EnumCalendarInfoCallback(string calendarString, IntPtr context)
         {
-            List<string> calendarDataList = (List<string>)((GCHandle)context).Target;
-            calendarDataList.Add(calendarString);
+            CallbackContext callbackContext = (CallbackContext)((GCHandle)context).Target;
+
+            if (callbackContext.DisallowDuplicates)
+            {
+                foreach (string existingResult in callbackContext.Results)
```

Comment:
The biggest issue here is that order matters, since the first element in the list is treated as the "default" and exposed in the API.  Using something like a dictionary isn't guaranteed to preserve the order they were added.
Also, a minor point, is that there are going to be less than 5 elements in the list, so the perf impact won't be that different between a List and a dictionary.